File: NeetCode/Sliding Window/LongestSubString.java
import java.util.HashSet;

public class LongestSubString {
    public static int lengthOfLongestSubString(String s) {
        int length = 0;
        int left = 0;
        char[] string = s.toCharArray();
        HashSet<Character> set = new HashSet<>();
        
        for(int right = 0;right < string.length;) {                         //! Time Complexity = O(n)
            while(set.contains(string[right])) {
                set.remove(string[left]);
                left += 1;
            }
            set.add(string[right]);
            right += 1;

            length = Math.max(length, right - left);
        }

        return length;
    }                                                                       //! Total Time Complexity = O(n)

    public static void main(String[] args) {
        String s = "abcabcbb";

        System.out.println(lengthOfLongestSubString(s));
    }
}

File: NeetCode/Sliding Window/LongestSubstring.cs
public class Solution
{
    public int LengthOfLongestSubstring(string s)
    {
        int left = 0;
        int right = 0;
        int max = 0;
        List<char> charList = new List<char>();

        if (string.IsNullOrEmpty(s))
        {
            return 0;
        }

        while (right < s.Length)
        {
            while (charList.Contains(s[right]))
            {
                charList.Remove(s[left]);
                left++;
            }

            charList.Add(s[right]);

            if (max < (right - left + 1))
            {
                max = right - left + 1;
            }

            right++;
        }

        return max;
    }
}
